lua/render-markdown/handler/markdown.lua

```diff
@@ -301,7 +301,11 @@ function Handler:language(code_block, add_background)
             -- Code blocks will pick up varying amounts of leading white space depending on the
             -- context they are in. This gets lumped into the delimiter node and as a result,
             -- after concealing, the extmark will be left shifted. Logic below accounts for this.
-            icon_text = str.pad(code_block.leading_spaces, icon_text .. info.text)
+            local pad = code_block.leading_spaces
+            if code.width == 'block' then
+                pad = code.language_pad
+            end
+            icon_text = str.pad(pad, icon_text .. info.text)
```

Comment:
You need to account for the leading spaces in both cases, otherwise `block` width will cut off the language hint when indented if no `language_pad` is specified. 

It should also not be tied to `block` width, this logic should work:

```lua
local icon_text = icon .. ' '
if self.context:hidden(info) then
  -- Code blocks will pick up varying amounts of leading white space depending on the
  -- context they are in. This gets lumped into the delimiter node and as a result,
  -- after concealing, the extmark will be left shifted. Logic below accounts for this.
  icon_text = str.pad(code_block.leading_spaces + code.language_pad, icon_text .. info.text)
end
return self:add(true, info.start_row, info.start_col, {
  virt_text = { { icon_text, highlight } },
  virt_text_pos = 'inline',
})
```

Reply:
[74fbd39](https://github.com/MeanderingProgrammer/render-markdown.nvim/pull/131/commits/74fbd3916115080e56d6134c81cffd4c2773fa66)